packages/components/src/components/select-wrapper/select-wrapper-dropdown/select-wrapper-dropdown-utils.spec.ts

```diff
@@ -339,6 +340,21 @@ describe('getValidOptions()', () => {
   });
 });
 
+describe('getLengthOfVisibleOptionsAndOptgroups()', () => {
+  it('should return number of options that are not hidden or initiallyHidden summed with number of optgroups', () => {
```

Comment:
I think we need a couple of more test cases for other scenarios like the one mentioned in the other comment.